prboom2/src/g_game.c

```diff
@@ -3512,21 +3575,175 @@ const byte* G_ReadDemoHeaderEx(const byte *demo_p, size_t size, unsigned int par
   demover = *demo_p++;
   longtics = 0;
 
+  // ano - jun2019 - special extensions. originally for UMAPINFO but
+  // designed to be extensible otherwise using the list of strings.
+  // note: i consulted the eternity engine implementation of this function
+  extension_version = -1;
   if (demover == 255)
   {
-	  // Uses UMAPINFO. The real version will be in the second byte.
-	  // This prepended 255 is here to prevent non-UMAPINFO ports from recognizing the demo.
-	  demover = *demo_p++;
-	  if (!umapinfo_loaded)
-	  {
-		  lprintf(LO_ERROR, "UMAPINFO not loaded but trying to play a demo recorded with it\n");
-	  }
-  }
-  else if (umapinfo_loaded)
-  {
-	  // Q: Should this abort?
-	  lprintf(LO_ERROR, "UMAPINFO loaded but trying to play a demo recorded without it\n");
+    int num_requirements;
+    // ano - jun2019
+    // so the format is
+    // demover byte == 255
+    // "PRB+" signature
+    // extension_version byte. for now this should always be "1"
+    // 1 byte for num_requirements
+
+    // num_requirements *
+    //    1 byte string length
+    //    and length chars (up to 255 obviously)
+    // note that the format has each length by each string
+    // as opposed to a table of lengths
+
+    // an example requirements string is "UMAPINFO".
+    // if num_requirements needs extending past 255 one day,
+    // consider reserving requirement string "MORE", which
+    // would could be implemented to load another list of 255.
```

Comment:
This might not ever be needed... but perhaps a better idea than adding a "MORE" string if the count goes past than 255 would be to add the next byte whenever a 255 is found ( so ``xFF x00`` would be 255 elements, ``xFF xFF 40" would be 550 elements, etc.)